src/lib/y2partitioner/dialogs/bcache.rb

```diff
@@ -0,0 +1,204 @@
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/popup"
+require "cwm/common_widgets"
+require "y2partitioner/dialogs/base"
+
+module Y2Partitioner
+  module Dialogs
+    # Form to set the backing and caching device for bcache.
+    # Part of {Actions::AddBcache}.
+    class Bcache < Base
+      # @param suitable_backing [Array<Y2Storage::BlkDevice>] devices that can be used for backing
+      # @param suitable_caching [Array<Y2Storage::BlkDevice,Y2Storage::BcacheCset>]
+      #   devices that can be used for caching
+      # @param device [Y2Storage::Bcache] device if already exists or nil for newly created one.
+      def initialize(suitable_backing, suitable_caching, device = nil)
+        textdomain "storage"
+        @caching = CachingDevice.new(device, suitable_caching)
+        @backing = BackingDevice.new(device, suitable_backing, @caching)
+      end
+
+      # @macro seeDialog
+      def title
+        _("Bcache Device")
+      end
+
+      # @macro seeDialog
+      def contents
+        VBox(
+          HBox(
+            @backing,
+            HSpacing(1),
+            @caching
+          ),
+          VSpacing(1)
+        )
+      end
+
+      # Selected caching device. Undefined if result of dialog is not `:next`
+      # @return [Y2Storage::BlkDevice, Y2Storage::BcacheCset]
+      def caching_device
+        @caching.result
+      end
+
+      # Selected backing device. Undefined if result of dialog is not `:next`
+      # @return [Y2Storage::BlkDevice]
+      def backing_device
+        @backing.result
+      end
+
+      # Widget to select the backing device
+      class BackingDevice < CWM::ComboBox
+        # @param device [Y2Storage::BlkDevice,nil] existing backing device or nil if it is new bcache
+        # @param devices [Array<Y2Storage::BlkDevice>] possible devices that can be used
+        #   as backing device for bcache
+        # @param caching [Y2Partitioner::Dialogs::Bcache::CachingDevice] reference to caching widget.
+        #   Used for validation if both caching and backing device is not identical.
+        def initialize(device, devices, caching)
+          textdomain "storage"
+          @device = device
+          @devices = devices
+          @caching = caching
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          _("Backing Device")
+        end
+
+        # @macro seeItemsSelection
+        def items
+          @devices.map do |dev|
+            [dev.sid.to_s, dev.name]
+          end
+        end
+
+        # @macro seeAbstractWidget
+        def help
+          # TRANSLATORS: %s stands for name of option
+          format(_(
+                   "<b>%s</b> is device that will be used as backing device for bcache." \
+                   "It will define available space of bcache. " \
```

Comment:
will define **the** available

or alternatively something more explicit like:

The resulting Bcache device will have the same size than this backing device.